src/charm.py

```diff
@@ -261,38 +261,71 @@ def _get_gateway_apis_kubernetes_resource_manager(self):
             logger=LOGGER,
         )
 
-    def _workload_tracing_config(self) -> Dict[str, str]:
-        """Set workload tracing configuration."""
-        tracing_config = {}
-        if not self.workload_tracing.is_ready():
-            return tracing_config
-
-        workload_tracing_endpoint = self.workload_tracing.get_endpoint("otlp_grpc")
-        if not workload_tracing_endpoint:
-            return tracing_config
-
-        tracing_config["meshConfig.enableTracing"] = "true"
-        tracing_config["meshConfig.extensionProviders[0].name"] = "otel-tracing"
-        tracing_config["meshConfig.extensionProviders[0].opentelemetry.port"] = urlparse(
-            f"//{workload_tracing_endpoint}"
-        ).port
-        tracing_config["meshConfig.extensionProviders[0].opentelemetry.service"] = urlparse(
-            f"//{workload_tracing_endpoint}"
-        ).hostname
-        tracing_config["meshConfig.defaultProviders.tracing[0]"] = "otel-tracing"
-        tracing_config["meshConfig.defaultConfig.tracing.sampling"] = 100.0
-
-        return tracing_config
+    def _workload_tracing_provider(self) -> Tuple[List[Any], Dict[str, Any]]:
+        """Return a tuple with a list containing the tracing provider and global tracing settings."""
+        providers = []
+        global_config = {}
+        if self.workload_tracing.is_ready():
+            workload_tracing_endpoint = self.workload_tracing.get_endpoint("otlp_grpc")
```

Comment:
nit: get_endpoint does an is_ready and returns None if not ready
```suggestion
        if workload_tracing_endpoint := self.workload_tracing.get_endpoint("otlp_grpc"):  # will be false if tracing is not ready because get_endpoint returns None
```

Reply:
Apparently we can't do this without a guard 

```
  File "/var/lib/juju/agents/unit-istio-k8s-0/charm/./src/charm.py", line 291, in _workload_tracing_provider
    if not (endpoint := self.workload_tracing.get_endpoint("otlp_grpc")):
  File "/var/lib/juju/agents/unit-istio-k8s-0/charm/lib/charms/tempo_coordinator_k8s/v0/tracing.py", line 931, in get_endpoint
    raise ProtocolNotRequestedError(protocol, relation)
charms.tempo_coordinator_k8s.v0.tracing.ProtocolNotRequestedError: ('otlp_grpc', None)
```

Reply:
hmm odd.  ok!